motionplan/tpspace/ptgIK.go

```diff
@@ -100,32 +100,29 @@ func NewPTGIK(simPTG PTG, logger logging.Logger, refDistLong, refDistShort float
 func (ptg *ptgIK) Solve(
 	ctx context.Context,
 	solutionChan chan<- *ik.Solution,
-	seed []referenceframe.Input,
+	seedVals []referenceframe.Input,
 	solveMetric ik.StateMetric,
 	nloptSeed int,
 ) error {
 	internalSolutionGen := make(chan *ik.Solution, 1)
 	defer close(internalSolutionGen)
 	var solved *ik.Solution
-	var gridSolved *ik.Solution
 
-	if seed == nil {
-		seed = ptg.defaultSeed
+	seed := make([]referenceframe.Input, 0, len(ptg.defaultSeed))
+	for _, defaultSeedVal := range ptg.defaultSeed {
```

Comment:
It seems unnecessary to deep copy this

Reply:
It is necessary; this function can be run in parallel on this struct, and if we don't deep copy then different threads will be editing one another's data structures.